source/common/network/happy_eyeballs_connection_impl.cc

```diff
@@ -490,6 +494,10 @@ void HappyEyeballsConnectionImpl::onEvent(ConnectionEvent event,
     ASSERT(connections_.size() == 1);
     // This connection attempt failed but there are no more attempts to be made, so pass
     // the failure up by setting up this connection as the final one.
+    ENVOY_CONN_LOG_EVENT(debug, "happy_eyeballs_cx_failed", "addresses={}", *this,
+                         address_list_.size());
+  } else {
```

Comment:
nit: it's invert the predicate in the if and move this to the top to keep both causes close to the conditional:

```
  if (event == ConnectionEvent::Connected) {
    ENVOY_CONN_LOG_EVENT(debug, "happy_eyeballs_cx_ok", "address={}", *this, next_address_);
  } else { 
    ENVOY_LOG(trace, "Connection failed to connect");
    ...
```